optimization/docs/pgbart_improvements.md

```diff
@@ -0,0 +1,111 @@
+# PG-BART Improvements
+
+This document outlines: (1) areas subject to performance and memory allocation improvements for the PyMC-BART codebase, and (2) potential methods to address these areas of improvement. First, an overview of the code profiling results is provided. Following the profiling results, several methods and languages will be discussed that could be utilized to improve the performance and memory allocation of the PyMC-BART codebase.
+
+## Code profiling results
+
+Code profiling was conducted using the `line_profiler` and `memray` packages for time and memory profiling, respectively. Below, a summary of the profiling results is provided. For specific details, refer to the `README.md` file within each `line_profiler` and `memray` directory of this module.
```

Comment:
I never tried, but I read in a newsletter this profiler: [scalene](https://github.com/plasma-umass/scalene)

I have that website bookmark for a few weeks and never test it, could be interesting for this project.

Reply:
Ahh very cool. Thanks for the link!